build.sbt

```diff
@@ -197,6 +197,7 @@ lazy val core = project
     AutomaticModuleName.settings("akka.stream.alpakka.kafka"),
     libraryDependencies ++= Seq(
         "com.typesafe.akka" %% "akka-stream" % akkaVersion,
+        "com.typesafe.akka" %% "akka-discovery" % akkaVersion % Provided,
```

Comment:
I guess it's fine to leave this as provided.  I tried the SNAPSHOT in a local app and could only get it to throw runtime exceptions when the user defined `DiscoverySupport.consumerBootstrapServers(consumerConfig)`, but didn't have the jar on the classpath, which seems like an obvious error on behalf of the user.  I'm not sure if it's worth handling this to present the user with a more instructive error.

FYI the exception was:

```
[error] (run-main-0) java.lang.NoClassDefFoundError: akka/discovery/Discovery$
java.lang.NoClassDefFoundError: akka/discovery/Discovery$
        at akka.kafka.scaladsl.DiscoverySupport$.bootstrapServers(DiscoverySupport.scala:38)
        at akka.kafka.scaladsl.DiscoverySupport$.bootstrapServers(DiscoverySupport.scala:58)
        at akka.kafka.scaladsl.DiscoverySupport$.$anonfun$consumerBootstrapServers$1(DiscoverySupport.scala:70)
        at akka.kafka.ConsumerSettings.enriched(ConsumerSettings.scala:599)
        at akka.kafka.ConsumerSettings.createKafkaConsumer(ConsumerSettings.scala:607)
        at com.seglo.SlowConsumer$.delayedEndpoint$com$seglo$SlowConsumer$1(SlowConsumer.scala:44)
        at com.seglo.SlowConsumer$delayedInit$body.apply(SlowConsumer.scala:21)
        at scala.Function0.apply$mcV$sp(Function0.scala:39)
        at scala.Function0.apply$mcV$sp$(Function0.scala:39)
        at scala.runtime.AbstractFunction0.apply$mcV$sp(AbstractFunction0.scala:17)
        at scala.App.$anonfun$main$1$adapted(App.scala:80)
        at scala.collection.immutable.List.foreach(List.scala:392)
        at scala.App.main(App.scala:80)
        at scala.App.main$(App.scala:78)
        at com.seglo.SlowConsumer$.main(SlowConsumer.scala:21)
        at com.seglo.SlowConsumer.main(SlowConsumer.scala)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
Caused by: java.lang.ClassNotFoundException: akka.discovery.Discovery$
        at java.net.URLClassLoader.findClass(URLClassLoader.java:382)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
        at akka.kafka.scaladsl.DiscoverySupport$.bootstrapServers(DiscoverySupport.scala:38)
        at akka.kafka.scaladsl.DiscoverySupport$.bootstrapServers(DiscoverySupport.scala:58)
        at akka.kafka.scaladsl.DiscoverySupport$.$anonfun$consumerBootstrapServers$1(DiscoverySupport.scala:70)
        at akka.kafka.ConsumerSettings.enriched(ConsumerSettings.scala:599)
        at akka.kafka.ConsumerSettings.createKafkaConsumer(ConsumerSettings.scala:607)
        at com.seglo.SlowConsumer$.delayedEndpoint$com$seglo$SlowConsumer$1(SlowConsumer.scala:44)
        at com.seglo.SlowConsumer$delayedInit$body.apply(SlowConsumer.scala:21)
        at scala.Function0.apply$mcV$sp(Function0.scala:39)
        at scala.Function0.apply$mcV$sp$(Function0.scala:39)
        at scala.runtime.AbstractFunction0.apply$mcV$sp(AbstractFunction0.scala:17)
        at scala.App.$anonfun$main$1$adapted(App.scala:80)
        at scala.collection.immutable.List.foreach(List.scala:392)
        at scala.App.main(App.scala:80)
        at scala.App.main$(App.scala:78)
        at com.seglo.SlowConsumer$.main(SlowConsumer.scala:21)
        at com.seglo.SlowConsumer.main(SlowConsumer.scala)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
```

Reply:
Not sure if that is a good idea. The error is obvious and the docs show that it needs to be added.

Reply:
Yes, I think the error is good enough as it is.

Reply:
We do similar in Akka when netty/aeron (optional dependencies) are not on the classpath: https://github.com/akka/akka/blob/master/akka-remote/src/main/scala/akka/remote/RemoteActorRefProvider.scala#L265

NoClassDef exceptions will scare some users

Reply:
Ok, added a handler for that.